src/commands/init.rs

```diff
@@ -99,28 +183,49 @@ pub fn init(cookbook_executable: &str, mut args: Peekable<Iter<String>>) -> Resu
             ))
         }
     };
-    let _shell_type = match args.next() {
+    let shell_env = match env::var("SHELL") {
+        Ok(val) => val,
+        Err(e) => return Err(format!("Unable to fetch ENV var $SHELL with error: {}", e)),
+    };
+    let shell_type = match args.next() {
         Some(arg) => arg,
         None => {
-            return Err(format!(
-                init_help_body!(),
-                "function name required for init invocation"
-            ))
+            // fish shell does not pass $0 as fish, fallback to reading $SHELL
+            if shell_env.contains("fish") {
+               "fish"
+            } else {
+                return Err(format!(
+                    init_help_body!(),
+                    "function name required for init invocation"
+                ))
+            }
         }
     };
     // Bootstrapping the sc section requires two parts:
     // 1. creating the function in question
     // 2. wiring up tab completion for the function.
     //
     // functions must be used instead of a script, as
-    // cookbook also supports commands that modify the
+    // tome also supports commands that modify the
     // current environment (such as cd you into a specific)
     // directory.
-    // TODO: add conditionals if other shells need different support
-    Ok(format!(
-        bash_zsh_init_body!(),
-        cookbook_executable = cookbook_executable,
-        script_root = script_root,
-        function_name = function_name
-    ))
+    match shell_type {
```

Comment:
This is verbose, but I couldn't figure out a more dynamic way to do so when the init_body!() executions are macros 🤔

Reply:
you could create a block just to extract the template, then pass that into the format argument. Something like:

```
let shell_template = match shell_type {
    "fish" => fish_init_body!(),
    ....,
    _ => {return Err()}
}

return Ok(format!(shell_template, tome_executable=tome_executable,...))  
```

Reply:
Of course :D I'll try that.

Reply:
I tried that approach and was having type issues 🤔. 

I can try again and then update with my findings.

Reply:
I wouldn't worry too much about it, it's a minor optimization that doesn't improve readability by that much, and correctness at all :) 

Thanks for trying!